lib/minify-css.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const CleanCSS = require('clean-css');
+const {wrapCSS, unwrapCSS} = require('./wrap-css');
+
+function createMinifyCSS(config) {
+	const cleanCSS = new CleanCSS(config);
+
+	return function (css, type) {
+		const wrappedCSS = wrapCSS(css, type);
+		const result = cleanCSS.minify(wrappedCSS);
+
+		if (result.warnings.length > 0 || result.errors.length > 0) {
+			return css;
```

Comment:
Shouldn't this handle failOnError / logOnError?  We should test this in the `errors` test.

Reply:
`html-minifier` only logs css minification errors (and ignores warnings, which is why we need to create our own version). Do you want to keep logging errors only, or add failure behavior too?

Reply:
I hadn't noticed before that `html-minifier` ignores warnings, I'm a concerned this adds unnecessary strictness by ignoring clean-css results when we get warnings but no errors.  I just did a local clone of `clean-css` and ran `git grep warnings -- lib/` to look at the warnings.  

I think `result.errors.length > 0` should respond to `failOnError` / `logOnError`.  I'm not sure what `results.warnings` should do.  Here's an example:
```css
.test {
  a: 1;
  background: blue;
}
```

The current version of this plugin will minify that to `.test{background:#00f}`.  `a` gets removed as it is an invalid property name which clean-css warns about.  This PR will result in the stylesheet not being minified at all.  I tested with `require('./lib/minify-css')({})('.test {background: blue;aa: 1}')` in REPL with the following modified code in minify-css.js:
```js
		if (result.warnings.length > 0 || result.errors.length > 0) {
			console.log({original: css, ...result});
			return css;
		}
```

This printed:
```
{
  original: '.test {background: blue;aa: 1}',
  styles: '.test{background:#00f}',
  stats: {
    efficiency: 0.2666666666666667,
    minifiedSize: 22,
    originalSize: 30,
    timeSpent: 27
  },
  errors: [],
  inlinedStylesheets: [],
  warnings: [ "Invalid property name 'aa' at 1:24. Ignoring." ]
}
```

Reply:
I hadn't considered the case of invalid/unknown property names. Indeed that's a big regression that's not suitable for everyone.

Unfortunately in the case of lit-element it's the only way to avoid mangling partial stylesheets, so it's the only way to get some real minification there :(

Does it still throw warnings when the option to remove invalid property is turned off? Personally I think we gain the most with just whitespace and newline removal.